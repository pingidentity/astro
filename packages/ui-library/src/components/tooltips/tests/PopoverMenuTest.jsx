window.__DEV__ = true;

jest.dontMock("../PopoverMenu");
jest.dontMock("../../../util/EventUtils.js");
jest.dontMock("../../../util/Utils.js");

describe("PopoverMenu", function() {
    var React = require("react"),
        ReactTestUtils = require("react-dom/test-utils"),
        TestUtils = require("../../../testutil/TestUtils"),
        PopoverMenu = require("../PopoverMenu");

    var buttonCallback = jest.fn();
    var buttonData = [
        {
            label: "First",
            onClick: buttonCallback
        },
        {
            label: "Second",
            onClick: buttonCallback
        },
        {
            label: "Third",
            onClick: buttonCallback
        },
        {
            label: "Fourth",
            onClick: buttonCallback
        }
    ];

    window.addEventListener = jest.fn();
    window.removeEventListener = jest.fn();
    beforeEach(function() {
        window.addEventListener.mockClear();
        window.removeEventListener.mockClear();
    });

    it("renders open state", function() {
        var component = ReactTestUtils.renderIntoDocument(
            <div>
                <PopoverMenu label="hello" open={true} buttons={buttonData} />
            </div>
        );
        var buttons = TestUtils.scryRenderedDOMNodesWithClass(
            component,
            "button-menu__button"
        );

        expect(buttons.length).toBe(4);
    });

    it("notifies button callback and toggle when clicking menu button", function() {
        var callback = jest.fn();

        var component = ReactTestUtils.renderIntoDocument(
            <div>
                <PopoverMenu
                    onToggle={callback}
                    label="hello"
                    open={true}
                    buttons={buttonData}
                />
            </div>
        );

        var button = TestUtils.scryRenderedDOMNodesWithClass(
            component,
            "button-menu__button"
        )[0];

        ReactTestUtils.Simulate.click(button, {});

        expect(buttonCallback).toBeCalled(); //make sure callback was triggered
        expect(callback).toBeCalled(); //make sure callback was triggered
    });

    it("doesn't error out if there's no event defined for a button", function() {
        var callback = jest.fn();

        var component = ReactTestUtils.renderIntoDocument(
            <div>
                <PopoverMenu
                    onToggle={callback}
                    label="hello"
                    open={true}
                    buttons={[{ label: "label" }]}
                />
            </div>
        );

        var button = TestUtils.scryRenderedDOMNodesWithClass(
            component,
            "button-menu__button"
        )[0];

        ReactTestUtils.Simulate.click(button, {});

        expect(callback).toBeCalled(); //make sure callback was triggered
    });
});
