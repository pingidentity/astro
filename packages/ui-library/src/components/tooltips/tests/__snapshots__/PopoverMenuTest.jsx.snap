// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverMenu data-id's don't change 1`] = `
Array [
  "popover-menu",
  "popover-menu-trigger",
  "popover-menu-button-first",
  "popover-menu-button-second",
  "popover-menu-button-third",
  "popover-menu-button-fourth",
]
`;
